app/src/index.js

```diff
@@ -1,29 +1,152 @@
+  projectId: "metrodreamin",
+  storageBucket: "metrodreamin.appspot.com",
+  messagingSenderId: "86165148906"
+};
+
+const stagingConfig = {
+  apiKey: "AIzaSyDYU-8dYy0OWGJ1RJ46V_S7fWJHlAA2DWg",
+  authDomain: "metrodreaminstaging.firebaseapp.com",
+  databaseURL: "https://metrodreaminstaging.firebaseio.com",
+  projectId: "metrodreaminstaging",
+  storageBucket: "metrodreaminstaging.appspot.com",
+  messagingSenderId: "572980459956"
+};
 
 export default function Index() {
+  const [user, setUser] = useState();
+  const [ database, setDatabase ] = useState();
+  const [ settings, setSettings ] = useState({ noSave: true });
+  const [ authListened, setAuthListened ] = useState(false);
+
+  useEffect(() => {
+    const useProd = determineIfProd();
+
+    firebase.initializeApp(useProd? prodConfig : stagingConfig);
+    setDatabase(firebase.firestore());
+
+    window.ui = new firebaseui.auth.AuthUI(firebase.auth());
+  }, []);
+
+  const signIn = (currentUser) => {
+    if (!currentUser || !currentUser.uid) return;
+
+    setSettings({
+      email: currentUser.email,
+      displayName: currentUser.displayName,
+      userId: currentUser.uid,
+      noSave: false
+    });
+
+    setUser(currentUser);
+
+    let userDoc = database.doc('users/' + currentUser.uid);
+    userDoc.get().then((doc) => {
+      if (doc) {
+        const data = doc.data();
+        if (data) {
+          let tempSettings = JSON.parse(JSON.stringify(settings));
+          for (const key in data) {
+            tempSettings[key] = data[key];
+          }
+          setSettings(prevSettings => {
+            return {...prevSettings, ...data};
+          });
+        }
+      }
+    }).catch((error) => {
+      console.log('Unexpected Error:', error);
+    });
+
+    userDoc.update({
+      lastLogin: Date.now()
+    }).then(() => {
+      // ReactGA.event({
+      //   category: 'User',
+      //   action: 'Signed In'
+      // });
+    }).catch((error) => {
+      console.log('Unexpected Error:', error);
+    });
+  }
+
+  const handleNoSave = () => {
+    setSettings(prevSettings => {
+      return {...prevSettings, ...{ noSave: true }};
+    });
+  }
+
+  const handleToggleTheme = (useLight) => {
+    setSettings(prevSettings => {
+      return {...prevSettings, ...{ lightMode: useLight }};
+    });
+  }
+
+  if (!authListened && database) {
+    firebase.auth().onAuthStateChanged((u) => {
+      if (u && u.uid) {
+        signIn(u);
+      }
+    });
+    setAuthListened(true);
+  }
+
   return (
     <Router>
       <Switch>
-        <Route exact path="/" children={<Parameterizer />} />
-        <Route path="/view/:viewIdEncoded?" children={<Parameterizer />} />
-        <Route exact path="/explore">
-          <h1>
-            ~Explore~
-          </h1>
-        </Route>
+        <Route exact path="/" children={<MainParameterizer
+                                          user={user}
+                                          database={database}
```

Comment:
look into seeing if we can solve these shared parameters in a more elegant way using useContent. maybe in future commit/pr